hack/jenkins/e2e-runner.sh

```diff
@@ -186,6 +179,37 @@ function dump_cluster_logs() {
     fi
 }
 
+# GCI specific settings.
+# Assumes: JENKINS_GCI_IMAGE_FAMILY
+function setup_gci_vars() {
+  local -r gci_staging_project=container-vm-image-staging
+  local -r image_info="$(gcloud compute images describe-from-family ${JENKINS_GCI_IMAGE_FAMILY} --project=${gci_staging_project} --format=json)"
+  local -r image_description="$(echo ${image_info} | jq '.description')"
+  local -r image_name="$(echo ${image_info} | jq '.name')"
+
+  if [[ "${JENKINS_USE_GCI_VERSION:-}" =~ ^[yY]$ ]]; then
+    # GCI QA jobs use the builtin k8s version.
+    # Staged GCI images all include versions in their image descriptions so we
+    # extract builtin Kubernetes version from them.
+    local -r k8s_version_re='.*Kubernetes: ([0-9a-z.-]+),.*'
+    if [[ ${image_description} =~ ${k8s_version_re} ]]; then
+      export JENKINS_PUBLISHED_VERSION="release/${BASH_REMATCH[1]}"
```

Comment:
Yes, I actually had a more reliable way implemented before I sent this PR
out. I think that one is going to take sometime to go through review, so
this can go in first to get us going. WDYT?

On Jul 27, 2016 5:10 PM, "Amey Deshpande" <notifications@github.com> wrote:

> In hack/jenkins/e2e-runner.sh
> <https://github.com/kubernetes/kubernetes/pull/29631#discussion_r72546726>
> :
>
> > @@ -186,6 +179,37 @@ function dump_cluster_logs() {
> >      fi
> >  }
> >
> > +# GCI specific settings.
> > +# Assumes: JENKINS_GCI_IMAGE_FAMILY
> > +function setup_gci_vars() {
> > +  local -r gci_staging_project=container-vm-image-staging
> > +  local -r image_info="$(gcloud compute images describe-from-family ${JENKINS_GCI_IMAGE_FAMILY} --project=${gci_staging_project} --format=json)"
> > +  local -r image_description="$(echo ${image_info} | jq '.description')"
> > +  local -r image_name="$(echo ${image_info} | jq '.name')"
> > +
> > +  if [[ "${JENKINS_USE_GCI_VERSION:-}" =~ ^[yY]$ ]]; then
> > +    # GCI QA jobs use the builtin k8s version.
> > +    # Staged GCI images all include versions in their image descriptions so we
> > +    # extract builtin Kubernetes version from them.
> > +    local -r k8s_version_re='.*Kubernetes: ([0-9a-z.-]+),.*'
> > +    if [[ ${image_description} =~ ${k8s_version_re} ]]; then
> > +      export JENKINS_PUBLISHED_VERSION="release/${BASH_REMATCH[1]}"
>
> IMO the image description is not the most reliable source to determine K8s
> version installed on the image. Did you have any other options in mind?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29631/files/081dfbbb7385a7bd20f629fbb3ed142ce3ed3ca8#r72546726>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGVaIX_EhswVMD6sJJsefNeQFF1i5gLZks5qZ_N3gaJpZM4JVkAr>
> .
>